Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -1536,158 +1566,121 @@ private void SetBoundingBoxCollider()
+            // Its possible we have two rigRoots here, the one about to be deleted and the new one
+            // Since those have the gizmo structure childed, be need to ommit them completely in the calculation of the bounds
+            // This can only happen by name unless there is a better idea of tracking the rigRoot that needs destruction
+
+            List<Transform> childTransforms = new List<Transform>();
+            childTransforms.Add(Target.transform);
+
+            foreach (Transform childTransform in Target.transform)
             {
-                bounds = GetSingleObjectBounds(Target);
-                boundsMethod = BoundsCalculationMethod.Collider;
-                return bounds;
+                if (childTransform.name.Equals(rigRootName)) { continue; }
+                childTransforms.AddRange(childTransform.GetComponentsInChildren<Transform>());
             }
-            else
-            {
-                for (int i = 0; i < toExplore.Count; ++i)
-                {
-                    var child = toExplore[i];
-                    if (bounds.size == Vector3.zero)
-                    {
-                        bounds = GetSingleObjectBounds(child.gameObject);
-                    }
-                    else
-                    {
-                        Bounds childBounds = GetSingleObjectBounds(child.gameObject);
-                        if (childBounds.size != Vector3.zero)
-                        {
-                            bounds.Encapsulate(childBounds);
-                        }
-                    }
-                }
 
-                if (bounds.size != Vector3.zero)
-                {
-                    boundsMethod = BoundsCalculationMethod.Colliders;
-                    return bounds;
-                }
-            }
+            // Iterate transforms and collect bound volumes
 
-            //simple case: sum of existing colliders
-            Collider[] colliders = Target.GetComponentsInChildren<Collider>();
-            if (colliders.Length > 0)
+            foreach (Transform childTransform in childTransforms)
             {
-                //Collider.bounds is in world space.
-                bounds = colliders[0].bounds;
-                for (int i = 0; i < colliders.Length; ++i)
+                if (childTransform == rigRoot) { continue; }
+
+                if (boundsCalculationMethod != BoundsCalculationMethod.RendererOnly)
                 {
-                    Bounds colliderBounds = colliders[i].bounds;
-                    if (colliderBounds.size != Vector3.zero)
-                    {
-                        bounds.Encapsulate(colliderBounds);
-                    }
+                    Collider collider = childTransform.GetComponent<Collider>();
+                    colliderByTransform = collider != null ? new KeyValuePair<Transform, Collider>(childTransform, collider) : new KeyValuePair<Transform, Collider>();
```

Comment:
KeyValuePair is a struct and I needed something to check for null in the code. But since I'm only to continue the loop in that case I can already do that here and just not create a KeyValuePair at all.